### PR TITLE
[analytics-node] Correct message callback type

### PR DESCRIPTION
**Pull request by nicklasmoeller**
analytics-node doesn't send pass any second argument to its callback for message methods.
It seems like it's never done so.

Callback construction happens at:
https://github.com/segmentio/analytics-node/blob/a78b2ac52a58a389e0880890b0e972290dab3a1a/index.js#L255

-----------------

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: 
  - https://github.com/segmentio/analytics-node/blob/a78b2ac52a58a389e0880890b0e972290dab3a1a/index.js#L255 (most recent published version)
  - https://github.com/segmentio/analytics-node/blob/bf5b2cda255266283a9be77cb125c80f777d5e1d/index.js#L226 (first commit)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing) - I adjusted the tests, to reflect the changes.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/analytics-node/analytics-node-tests.ts [hunk 1]
```diff
@@ -40,11 +40,9 @@ function testIdentify(): void {
       plan: 'Enterprise',
       friends: 42
     }
-  }, (err, data) => {
```
Comment (thomasthiebaud) [inline types/analytics-node/analytics-node-tests.ts:43]: It does pass `data` to the callback https://github.com/segmentio/analytics-node/blob/a78b2ac52a58a389e0880890b0e972290dab3a1a/index.js#L256

**Review by thomasthiebaud — CHANGES_REQUESTED** [review]

File: types/analytics-node/analytics-node-tests.ts (hunk 1 above)
Comment (nicklasmoeller) [reply types/analytics-node/analytics-node-tests.ts:43]: @thomasthiebaud that's the flush callback, not the message callback. The flush signature is untouched.

Feel free to actually test this. Only an error argument is passed for messages.

**Comment by nicklasmoeller** [comment]
Although I've commented on the review, I'm gonna address this as a direct comment, since that's persisted better in Github's UI.

There are two different types of callbacks in `analytics-node`. You'll see them both being executed here: https://github.com/segmentio/analytics-node/blob/a78b2ac52a58a389e0880890b0e972290dab3a1a/index.js#L255-L256:

The first line, iterates over `callbacks`, which is all the messages that have been published by the `enqueue` method (happens behind the scenes, but `.identify`, `.track`, etc.
The next line is `callback`. That's the argument provided for `.flush`.

The first line (callbacks for messages) is only provided `(err)`, whilst the second is provided `(err, data)`.

The changes applied in this PR only alters the signature for the callbacks provided to messages.

I hope this clears up any confusion.

**Comment by thomasthiebaud** [comment]
@nicklasmoeller Thanks for the details. Then it looks good. I'll dismiss my review

**Review by thomasthiebaud — APPROVED** [review]

**Comment by nicklasmoeller** [comment]
:+1: 

Ready to merge

**Comment by nicklasmoeller** [comment]
Ready to merge

**Comment by benjaminhoffman** [comment]
hi @nicklasmoeller im late to the party on this one, sorry, but hit this issue while debugging my Segment analytics. I read your comments in this PR thread so maybe I'm missing something but I'm fairly certain that `analytics-node` callback _does in fact_ take two parameters (err, batch).  I'm okay being wrong but here is why I believe this:

1) [their docs say so](https://segment.com/docs/connections/sources/catalog/libraries/server/node/#configuration) (see code snippet of the **Batching** section)

2) I tested it (see attached)

I think the `callback` argument to track/identify/etc events is a flush callback that takes two arguments... but maybe I'm missing something? Just strange to me that it's working locally.

Screen Shot 2022-02-07 at 21 16 00

**Comment by benjaminhoffman** [comment]
Follow up, here is the PR that Segment merged that introduces a second param to the callback: https://github.com/segmentio/analytics-node/pull/315/files

**Pull request merged**